tests/cpp/tree/test_quantile_hist.cc

```diff
@@ -21,23 +21,25 @@ namespace tree {
 class QuantileHistMock : public QuantileHistMaker {
   static double constexpr kEps = 1e-6;
 
-  struct BuilderMock : public QuantileHistMaker::Builder {
-    using RealImpl = QuantileHistMaker::Builder;
+  template <typename GradientSumT>
+  struct BuilderMock : public QuantileHistMaker::Builder<GradientSumT> {
```

Comment:
I initiated this test mock back then because quantile hist was an integrated class with very little outside dependency.  Now it's growing thanks to your contributions.  ;-)  You can consider splitting it up in the future.

Reply:
ok. But now only type was added :)